pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -250,324 +252,219 @@ func TestReconcile(t *testing.T) {
-							Name: "workspace",
-							Type: v1beta1.PipelineResourceTypeGit,
-						},
-					}},
-				},
-			},
-		},
-	}
-	clusterTasks := []*v1beta1.ClusterTask{
-		{
-			ObjectMeta: metav1.ObjectMeta{Name: "unit-test-cluster-task"},
-			Spec: v1beta1.TaskSpec{
-				Params: []v1beta1.ParamSpec{
-					{
-						Name: "foo",
-						Type: v1beta1.ParamTypeString,
-					},
-					{
-						Name: "bar",
-						Type: v1beta1.ParamTypeString,
-					},
-					{
-						Name: "templatedparam",
-						Type: v1beta1.ParamTypeString,
-					},
-					{
-						Name: "contextRunParam",
-						Type: v1beta1.ParamTypeString,
-					},
-					{
-						Name: "contextPipelineParam",
-						Type: v1beta1.ParamTypeString,
-					},
-				},
-				Resources: &v1beta1.TaskResources{
-					Inputs: []v1beta1.TaskResource{{
-						ResourceDeclaration: v1beta1.ResourceDeclaration{
-							Name: "workspace",
-							Type: v1beta1.PipelineResourceTypeGit,
-						},
-					}},
-					Outputs: []v1beta1.TaskResource{
-						{
-							ResourceDeclaration: v1beta1.ResourceDeclaration{
-								Name: "image-to-use",
-								Type: v1beta1.PipelineResourceTypeImage,
-							},
-						},
-						{
-							ResourceDeclaration: v1beta1.ResourceDeclaration{
-								Name: "workspace",
-								Type: v1beta1.PipelineResourceTypeGit,
-							},
-						},
-					},
-				},
-			},
-		},
-		{
-			ObjectMeta: metav1.ObjectMeta{Name: "unit-test-followup-task"},
-			Spec: v1beta1.TaskSpec{
-				Resources: &v1beta1.TaskResources{
-					Inputs: []v1beta1.TaskResource{{
-						ResourceDeclaration: v1beta1.ResourceDeclaration{
-							Name: "workspace",
-							Type: v1beta1.PipelineResourceTypeGit,
-						},
-					}},
-				},
-			},
-		},
+
+	prs := []*v1beta1.PipelineRun{
+		parse.MustParsePipelineRun(t, fmt.Sprintf(`
+metadata:
+  name: %s
+  namespace: foo
+spec:
+  pipelineRef:
+    name: test-pipeline
+  serviceAccountName: test-sa
+  resources:
+    - name: git-repo
+      resourceRef:
+        name: some-repo
+    - name: best-image
+      resourceSpec:
+        type: %s
+        params:
+          - name: url
+            value: gcr.io/sven
+  params:
+    - name: bar
+      value: somethingmorefun
+`, pipelineRunName, resourcev1alpha1.PipelineResourceTypeImage)),
 	}
-	rs := []*resourcev1alpha1.PipelineResource{{
-		ObjectMeta: baseObjectMeta("some-repo", "foo"),
-		Spec: resourcev1alpha1.PipelineResourceSpec{
-			Type: resourcev1alpha1.PipelineResourceTypeGit,
-			Params: []resourcev1alpha1.ResourceParam{{
-				Name:  "url",
-				Value: "https://github.com/kristoff/reindeer",
-			}},
-		},
-	}}
+
+	ps := []*v1beta1.Pipeline{
+		parse.MustParsePipeline(t, fmt.Sprintf(`
+metadata:
+  name: %s
+  namespace: foo
+spec:
+  resources:
+    - name: git-repo
+      type: %s
+    - name: best-image
+      type: %s
+  params:
+    - name: pipeline-param
+      default: somethingdifferent
+      type: %s
+    - name: rev-param
+      default: revision
+      type: %s
```

Comment:
`type` here and on line 296 are not necessary after https://github.com/tektoncd/pipeline/pull/4608 is merged.

Without specifying `type` here, the test fails with:

```
logger.go:130: 2022-02-18T10:13:50.249-0800	ERROR	TestReconcile	pipelinerun/pipelinerun.go:228	Reconcile error: "" type does not match default value's type: "string": finally.params[pipeline-param].default.type, finally.params[pipeline-param].type, finally.params[rev-param].default.type, finally.params[rev-param].type, tasks.params[pipeline-param].default.type, tasks.params[pipeline-param].type, tasks.params[rev-param].default.type, tasks.params[rev-param].type
```

https://github.com/tektoncd/pipeline/pull/4608 is fixing this bug.